MicroServices/ShopFlowPro-service/src/main/java/com/wsu/shopflowproservice/controller/ServiceOrderController.java

```diff
@@ -0,0 +1,75 @@
+package com.wsu.shopflowproservice.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.wsu.shopflowproservice.dto.ServiceOrderDTO;
+import com.wsu.shopflowproservice.dto.ServiceResponseDTO;
+import com.wsu.shopflowproservice.service.ServiceOrderService;
+
+import io.micrometer.core.ipc.http.HttpSender.Response;
+import lombok.RequiredArgsConstructor;
+
+import static com.wsu.shopflowproservice.utilities.Constants.MESSAGE;
+import static com.wsu.shopflowproservice.utilities.Constants.PAGE_COUNT;
+import static com.wsu.shopflowproservice.utilities.Constants.RESULT_COUNT;
+
+import java.util.Map;
+
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/serviceOrders")
+public class ServiceOrderController {
+
+    private final ServiceOrderService serviceOrderService;
+
+    /**
+     * This endpoint is used to get details of serviceOrder via the serviceOrderid
+     * @param serviceOrderId - id used to get serviceORder details
+     * @return - ServiceResponseDTO which contains details of serviceOrder
+     */
+    @GetMapping(value = "{serviceOrderId}")
```

Comment:
So would this be another GET method within the ServiceOrderController with similar components just passing different variables?

Reply:
We given example to have summary endpoint in new Controller class because joining many tables and getting summarized data in response for work order. 
If you want to achieve same then you can have in Separate controller & service class for better readability, otherwise you can add new GET endpoint in same controller class also and use JPA method to return ServiceOrder paginated response by applying filter if any.